src/main/java/codeu/model/store/basic/ConversationStore.java

```diff
@@ -117,6 +117,15 @@ public void setConversations(List<Conversation> conversations) {
     this.conversations = conversations;
   }
 
+  /** Return latest conversation created */
+  public Conversation getLatestConversation(){
+    if(conversations.size() > 0) {
```

Comment:
I'm not sure you can assume conversations isn't null here..

And remember, this class isn't threadsafe. Strictly speaking all these methods should probably be synchronized. conversations could change between the 3 calls, so beware. Doesn't matter much here but in a real application you'd definitely need to anticipate that. At a minimum it would be customary to add a comment in the class javadoc to point that out.

Also, it's a bit more polished isEmpty() instead of size(), but it doesn't matter.

Same comment for all the others like this

Reply:
Just fixed the null assumption! I didn't knew it wasn't threadsafe in fact I know very little about it, it's logic what you are saying that the values can change between the 3 calls, I will investigate and inform myself better about these type of problems and how to address them correctly!